packages/cozy-stack-client/src/CozyStackClient.spec.js

```diff
@@ -597,6 +597,41 @@ describe('CozyStackClient', () => {
       const newToken = client.refreshToken()
       await expect(newToken).rejects.toThrow()
     })
+
+    fit(`should NOT try to refresh several times in parallele`, async () => {
+      const client = new CozyStackClient(FAKE_INIT_OPTIONS)
+      jest.spyOn(client, 'refreshToken')
+      jest.spyOn(client, 'fetch')
+      global.fetch.mockResponse(() => {
+        return Promise.resolve({
+          headers: {
+            'content-type': 'application/json; charset=UTF-8'
+          },
+          body: JSON.stringify({
+            error: 'Expired token'
+          }),
+          ok: false,
+          status: 403,
+          statusText: '',
+          type: 'default',
+          url: 'http://cozy.tools:8080/settings/capabilities'
+        })
+      })
+
+      try {
+        await Promise.all([
+          client.fetchJSON('GET', '/test'),
+          client.fetchJSON('GET', '/test')
+        ])
+      } catch (e) {
+        expect(e).toBeInstanceOf(FetchError)
+        expect(client.fetch).toHaveBeenCalledTimes(2)
+
+        expect(client.refreshToken).toHaveBeenCalledTimes(1)
+        const token = client.getAccessToken()
+        expect(token).toEqual(FAKE_INIT_OPTIONS.token)
```

Comment:
I would move those 3 lines outside of the `catch`, to be sure to not have a green test only because we don't raise an exception on ` await Promise.all` (lines 622-625).

Reply:
done